wallet/createtx.go

```diff
@@ -206,6 +215,14 @@ func (w *Wallet) txToOutputs(outputs []*wire.TxOut,
 			// according to the selected coin selection strategy.
 			wrappedEligible := make([]Coin, len(eligible))
 			for i := range eligible {
+				// Restrict the selected utxos if a filter
+				// function is provided.
+				if allowUtxo != nil &&
+					!allowUtxo(eligible[i]) {
+
+					continue
+				}
+
```

Comment:
Maybe we can move this check to `findEligibleOutputs` function?  Seems inline with the function definition and prevents this duplicity?